tqdm/_tqdm.py

```diff
@@ -41,12 +43,75 @@ def __init__(self, msg, fp_write=None, *a, **k):
             super(TqdmDeprecationWarning, self).__init__(msg, *a, **k)
 
 
+class TMonitor(Thread):
+    """ tqdm monitoring thread
+    Monitors if tqdm bars are taking too much time to display
+    and readjusts miniters automatically if necessary"""
+
+    # internal vars for unit testing
+    _time = None
+    _sleep = None
+
+    def __init__(self, tqdm_cls, sleep_interval):
+        Thread.__init__(self)
+        self.exit_event = Event()
+        self.daemon = True  # kill thread when main killed (KeyboardInterrupt)
+        self.was_killed = False
+        self.tqdm_cls = tqdm_cls
+        self.sleep_interval = sleep_interval
+        if TMonitor._time is not None:
+            self._time = TMonitor._time
+        else:
+            self._time = time
+        if TMonitor._sleep is not None:
+            self._sleep = TMonitor._sleep
+        else:
+            self._sleep = sleep
+        self.start()
+
+    def exit(self):
+        self.exit_event.set()
+        self.was_killed = True
+        # self.join()  # DO NOT, blocking event, slows down tqdm at closing
```

Comment:
To avoid anyone reimplementing it later. That's useful commented code to avoid regressions.